lib/os/p4wq.c

```diff
@@ -19,8 +19,11 @@ struct device;
 static void set_prio(struct k_thread *th, struct k_p4wq_work *item)
 {
 	__ASSERT_NO_MSG(!IS_ENABLED(CONFIG_SMP) || !z_is_thread_queued(th));
+	
+	k_timeout_t dl = K_CYC(item->deadline);
+
 	th->base.prio = item->priority;
-	th->base.prio_deadline = item->deadline;
+	th->base.prio_deadline = CLAMP(((int32_t)dl.ticks), 0, INT_MAX);
```

Comment:
not really a part of this PR, but I thought it would be best to format it according to the new API.